### DISCUSSION TITLE
next.config.js dont change path

### DISCUSSION THREAD

**Discussion by edanTavory** [vercel/next.js#37583] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

next 11.14
and i change from last version because its not work there

### What browser are you using? (if relevant)

chrome

### Describe the Bug

next.config.js dont change path 

### Expected Behavior

the path need to change name

### To Reproduce

module.exports = {
  async rewrites() {
    console.log("Rewrites called");
    return [
      {
        source: "/heatPumps",
        destination: "/משאבות-חום",
      },
      {
        source: "/underFlooorHeat",
        destination: "/חימום-תת-רצפתי",
      },
    ];
  },
};

**Comment by icyJoseph** [comment]
Are you expecting calls to `/heatPumps` to show in the browser as `/משאבות-חום`, then you should use `redirects`, not `rewrites`.

Rewrites keep the URL in the browser (in this case `heatPumps`) but render the content under `משאבות-חום`

**Comment by edanTavory** [comment]
module.exports = {
  async redirects() {
    console.log("Rewrites called");
    return [
      {
        source: "/heatPumps",
        destination: "/משאבות-חום",
        permanent: true,
        
      },
      {
        source: "/underFlooorHeat",
        destination: "/חימום-תת-רצפתי",
        permanent: true,

      },
    ];
  },
};




now its work but get (bad request to the page)
its change the url but no see page

**Reply by edanTavory** [reply]
i just want to change name of path to hebrow not the page

**Reply by icyJoseph** [reply]
I am not sure if this entirely possible. It is possible to do this client side but it will defeat the purpose of SEO and SSR.

I personally think that the `next.config.js` might not be the best place to try and to do this:

I'd do a `_middleware`, which works even outside Vercel, like this:

```ts
import { NextResponse, type NextRequest } from "next/server";

export function middleware(req: NextRequest) {
  const { pathname } = req.nextUrl;

  if (
    pathname.startsWith("/api") || //  exclude all API routes
    pathname.startsWith("/static") || // exclude static files
    pathname.includes(".") // exclude all files in the public folder
  )
    return NextResponse.next();

  if (pathname === "/item") {
    req.nextUrl.pathname = "/translated";
    return NextResponse.redirect(req.nextUrl);
  }

  if (pathname === "/translated") {
    req.nextUrl.pathname = "/item";

    return NextResponse.rewrite(req.nextUrl);
  }

  return NextResponse.next();
}
```

With this file structure:

```
pages/
├── _app.tsx
├── _middleware.ts
├── api
├── index.tsx
└── item.tsx
```

When I navigate to `/item` I get sent to `/translated`, with the contents of `pages/item`

**Reply by edanTavory** [reply]
i need that for seo in hebrow languish,
so i want all the web page in hebrow so i need way to change,
because all the name of docs need to be in english.
so i dont want translate i just want other langwish web

**Reply by edanTavory** [reply]
hey?
